### ISSUE TITLE
Chunker fails to convert certain Wall types, as well as Barriers

### ISSUE THREAD

**Issue by KadTheHunter** [HiveGamesOSS/Chunker#40]
### Chunker Version

1.1.1-main-79e9191

### Input World Version

Bedrock 1.21.20

### Output World Version

Java 1.21.0

### Platform / Operating System

Windows 10

### Issue Description

When converting my world, Chunker fails to convert certain wall types (two confirmed being Diorite and Stone Brick), leaving air instead.

I tried to create a block mapping for it, only to find that neither Diorite nor Stone Brick (and seemingly any pre-1.16 Wall) was an option in the Bedrock dropdown list, only post-1.16 Walls like Blackstone, Deepslate, Mud, Tuff, etc.

Additionally, it does not convert Bedrocks `invisible_bedrock` to Javas `barrier`, though I am able to set up a block mapping for that.

*Additionally*, it seems plants inside flower pots do not convert, though the flower pots themselves do. 

EDIT:
Here is the map file used:
[world.zip](https://github.com/user-attachments/files/16982036/world.zip)

**Comment by FormallyMyles** [comment]
I looked through your world and even though it is technically a 1.21 Bedrock world, it has a lot of old data from prior to 1.8 MCBE which Chunker doesn't support and also a lot of invalid block entities etc. I would also guess that your flower pots are storing the flower data in the old format and don't convert because of this.

I'm not entirely sure regarding the block-entity data but when you edit chunks in Bedrock they do upgrade the format of the chuk so you may find that helps with some of the issues.

In regards to your block mapping query, in 1.21.20 the walls you're looking for are under cobblestone_wall as the wall_type property which in later versions (1.21.30) do become their own names in Bedrock. 

If you would like to give co-ordinates to an example bit of wall that is missing, I can look but I would guess it might be related to the world having old data.

**Comment by KadTheHunter** [comment]
Interesting!

The world itself originally comes from a LEET.cc server (LEET recently shut down and provided all owners with a copy of their maps), I took the files and loaded them into a PocketMine-MP 5.0+ server to upgrade/convert them to the leveldb format, and then finally loaded it in singleplayer to generate the level.dat.

I guess LEET was somehow saving the worlds in pre-1.8 format, running on PM3, and supporting the latest MCPE versions. Whacky.

I'll try the chunk update method and see how it goes then.

Coordinates for example walls:
- Stone Brick Walls:
  - `10006 64 -21` (the lamppost)
- Diorite Wall:
  - `9986 64 -78` (porch pillars)
- Cobblestone Wall
  - `9940 66 -197` (porch pillar bases)
- Andesite Wall
  - `9968 64 14` (bus stop corners)

**Comment by KadTheHunter** [comment]
Ok, custom mapping the walls seems to have worked!

Flowerpots contents still don't transfer unless I update them first in Bedrock, which is probably related to the pre-1.8 data you mentioned. Will try chunk updates and see if that helps.

**Comment by FormallyMyles** [comment]
When I looked on your world converted to 1.21.1 the stone brick walls on the lamp post seemed to look fine. I don't suppose you have screenshots of what you're seeing?

Additionally do you have co-ordinates / screenshots for the location of the invisible_bedrock? Chunker should have mappings for that to turn into barriers for Java.

**Comment by KadTheHunter** [comment]
The invisible_bedrock is straight above the coordinates provided above, it serves as a wall/ceiling around the entire build.

I've just tried the process once more using a fresh copy of the map from a backup, updated chunks around 9986 / -42, converted with no custom mappings, and everything seems to have ported fine. Item Frames/Flower Pots in the updated chunks still have their contents while those outside the updated chunks don't, but walls and barriers everywhere ported fine.

I'm no longer sure what exactly is happening, given I started with the same fresh copy 2 days ago (using Chunker 1.1.2) and had to custom map walls and barriers to get them to convert.

Going to close this issue, as it seems the walls/barriers *do* port correctly, again no clue what was happening 2 days ago.

Only remaining problem is block entities not retaining their data/contents, which is apparently due to the pre-1.8 data and solved with chunk updates; Thank you for suggesting that.

Apologies for any time wasted.
